apollo-router/src/services/layers/query_analysis.rs

```diff
@@ -51,8 +50,8 @@ impl QueryAnalysisLayer {
         }
     }
 
-    pub(crate) fn make_compiler(&self, query: &str) -> (ApolloCompiler, FileId) {
-        Query::make_compiler(query, self.schema.api_schema(), &self.configuration)
+    pub(crate) fn parse_document(&self, query: &str) -> ParsedDocument {
+        Query::parse_document(query, self.schema.api_schema(), &self.configuration)
```

Comment:
I would love to see a future where we also don't reparse operations. Given how incredibly large some operations can be, a reparse would be significant. Perhaps a problem for future us.

Reply:
This PR already eliminates the serialize + reparse step at the end of every query transform visitor. There’s one maybe-unnecessary maybe-reparse in `AuthorizationPlugin::filter_query` tracked in https://github.com/apollographql/router/issues/4040, but other than that this function is used for the initial parse.

Reply:
gotcha!